config/src/test/java/org/springframework/security/config/annotation/web/configuration/WebSecurityConfigurationTests.java

```diff
@@ -323,7 +324,7 @@ public void loadConfigWhenTwoSecurityFilterChainsPresentAndSecondWithAnyRequestT
 		assertThatExceptionOfType(BeanCreationException.class)
 			.isThrownBy(() -> this.spring.register(MultipleAnyRequestSecurityFilterChainConfig.class).autowire())
 			.havingRootCause()
-			.isExactlyInstanceOf(IllegalArgumentException.class);
+			.isExactlyInstanceOf(UnreachableFilterChainException.class);
```

Comment:
While this is more accurate of a test at this point, please leave tests that don't have to be modified as-is.

Reply:
Here we will have to use the `isInstanceOf` check, because `isExactlyInstanceOf` uses a check like:
```
expectedType.equals(actual.getClass()) 
```
and it will not succeed even if `UnreachableFilterChainException` extends `IllegalArgumentException`.

Reply:
Hmm, I see, so it needs to change either way. Let's please change it to `.isInstanceOf` to make it more resilient to this kind of change.